system/lib/libc/emscripten_syscall_stubs.c

```diff
@@ -42,15 +43,22 @@ static mode_t g_umask = S_IRWXU | S_IRWXG | S_IRWXO;
     return -ENOSYS; \
   }
 
+#define STRINGIFY(s) #s
+#define STR(s) STRINGIFY(s)
```

Comment:
Is this indirection through a second macro important? Can we get away with just one macro instead?

Reply:
I assume it is necessary in order to stringify the macro expansion of the argument instead of the argument itself. So without the indirection it would create `"__EMSCRIPTEN_major__"` but with the indirection it instead creates `"<contents of __EMSCRIPTEN_major__>"`. This sort of thing is very typical of macro languages.

Reply:
Yes this is a strange artifact of the C pre-processor: https://gcc.gnu.org/onlinedocs/gcc-4.8.5/cpp/Stringification.html "If you want to stringify the result of expansion of a macro argument, you have to use two levels of macros."

Reply:
:cry: